autogen/variables.tf.tmpl

```diff
@@ -85,65 +85,65 @@ variable "backends" {
   description = "Map backend indices to list of backend maps."
   type = map(object({
     {% if not serverless %}{# not necessary for serverless as default port_name=http, protocol=HTTP #}
```

Comment:
If the fields are all optional now, do we need this template check? Does it matter if this var exists in all modules generated for this template?

Reply:
Although fields are optional now but I am worried people will provide value and get error. In order to avoid it I just wanted to remove it from Serverless NEG. Since these are all optional people can use main module to deploy serverless NEG.